src/components/common/FieldForm.js

```diff
@@ -0,0 +1,187 @@
+import {
+  Button,
+  FormControl,
+  FormGroup,
+  InputLabel,
+  MenuItem,
+  Select,
+  Snackbar,
+  TextField,
+  Typography,
+} from '@material-ui/core';
+import { makeStyles } from '@material-ui/core/styles';
+import { Alert } from '@material-ui/lab';
+import { createField, updateField } from 'dataclients/FieldsClient';
+import { getLocations } from 'dataclients/LocationsClient';
+import React, { useEffect, useState } from 'react';
+
+const useStyles = makeStyles((theme) => ({
+  formControl: {
+    marginTop: theme.spacing(1),
+    marginBottom: theme.spacing(1),
+    marginLeft: 'auto',
+    marginRight: 'auto',
+    minWidth: 300,
+    maxWidth: 300,
+  },
+  submitButton: {
+    margin: theme.spacing(3),
+    width: 'fit-content',
+    minWidth: '150px',
+    marginLeft: 'auto',
+    marginRight: 'auto',
+  },
+  menuItem: {
+    maxWidth: 300,
+    whiteSpace: 'normal',
+  },
+  selectEmpty: {
+    marginTop: theme.spacing(2),
+  },
+}));
+
+const FieldForm = ({ operation, title, selectedRow, closeHandler }) => {
+  const classes = useStyles();
+  const [locality, setLocality] = useState(selectedRow.locationId);
+  const [fieldIdentifier, setFieldIdentifier] = useState(selectedRow.name);
+  const [area, setArea] = useState(selectedRow.area);
+  const [locations, setLocations] = useState([]);
+  const [alertMessage, setAlertMessage] = useState('');
+  const [alertStatus, setAlertStatus] = useState(false);
+  const [alertSeverity, setAlertSeverity] = useState('');
+
+  useEffect(() => {
+    getLocations()
+      .then(setLocations)
+      .catch((e) => {
+        console.log('Fetching location list failed', e);
+        showAlert('Fetching location list failed', 'error');
+      });
+  }, []);
+
+  const handleAlertClose = (_event, reason) => {
+    if (reason === 'clickaway') {
+      return;
+    }
+    setAlertStatus(false);
+  };
+
+  const showAlert = (message, severity) => {
+    setAlertMessage(message);
+    setAlertSeverity(severity);
+    setAlertStatus(true);
+  };
+
+  const handleTextChange = ({ target }) => {
+    const { name, value } = target;
+    if (name === 'fieldId') {
+      setFieldIdentifier(value);
+    } else if (name === 'area') {
+      setArea(value);
+    }
+  };
+
+  const handleSubmit = () => {
+    let payload = {};
+    // payload.name = fieldIdentifier;
+    payload.identifier = fieldIdentifier;
+    payload.location = locality;
+    payload.area = Number(area);
+    payload.id = selectedRow.obj.id;
+    
+    // console.log(selectedRow.obj);
+    if (operation === 'UPDATE') {
+      updateField(payload)
+        .then((response) => {
+          // showAlert('Field successfully updated', 'info');
+          closeHandler();          
+        })
+        .catch((e) => {
+          console.log('Internal server error', e);
+          showAlert('Field updation failed: ' + e.message, 'error');
+        });
+    } else {
+      createField(payload)
```

Comment:
Let's remove this for now and add later if required/used.

Reply:
will be needed soon, since add field will use the same form